Scripts For backup

bourasom opened the pull request.

bourasom committed:
backup: general script

bourasom committed:
Stop eXo and ES

bourasom committed:
Mongo and db start/stop

bourasom committed:
backup database

bourasom committed:
backup data and mongodb

vsellier commented on srv/exo/bin/_setenv.sh:
```diff
@@ -0,0 +1,38 @@
+#!/bin/sh -ue
```
You should rename this file in _setenv-template.sh to show it must be renamed and changed according the environment specificities

vsellier commented on srv/exo/bin/_setenv.sh:
```diff
@@ -0,0 +1,38 @@
+EXO_MONGO_SERVER=${MONGO_SERVER}
+EXO_USER=exo
+#Database Backup
+DB_DUMP_WORKING_DIR=/srv/backup
```
You should not work directly on the final backup directory because you will erase the previous backup at the beginning. If several services are on the same server the last backuped service will erase the other service backup

vsellier commented on srv/exo/bin/_dumpDatabase.sh:
```diff
@@ -0,0 +1,29 @@
+echo "[INFO] = Dumping database ${PLF_DATABASE_NAME} into ${DB_DUMP_WORKING_DIR} ..."
+echo "[INFO] ======================================="
+
+
```
```suggestion
```

vsellier commented on srv/exo/bin/_dumpDatabase.sh:
```diff
@@ -0,0 +1,29 @@
+echo "[INFO] ======================================="
+echo "[INFO] = Dumping database ${PLF_DATABASE_NAME} into ${DB_DUMP_WORKING_DIR} ..."
+echo "[INFO] ======================================="
+
```
```suggestion
echo "[INFO] $(display_date)"

```

vsellier commented on srv/exo/bin/_dumpDatabase.sh:
```diff
@@ -0,0 +1,29 @@
+
+display_time sudo mysqldump ${DUMP_OPTIONS} ${PLF_DATABASE_NAME} > ${DB_DUMP_WORKING_DIR}/${PLF_DATABASE_NAME}-dumpDataBase.sql
+
+echo "[INFO] Done"
```
```suggestion
echo "[INFO] $(display_date)"
echo "[INFO] Done
```

vsellier commented on srv/exo/bin/_dumpDatabase.sh:
```diff
@@ -0,0 +1,29 @@
+
+echo "[INFO] Done"
+
+popd > /dev/null 2>&1
```
```suggestion
popd > /dev/null 2>&1

```

vsellier commented on srv/exo/bin/_dumpData.sh:
```diff
@@ -0,0 +1,27 @@
+display_time tar --directory ${DATA_WORKING_DIR} --use-compress-prog=pbzip2 -cpf ${PLF_BCK_DIR}/${PLF_NAME}-data-${SCRIPT_DATE}.tar.bz2 data  
+pushd ${PLF_BCK_DIR} > /dev/null 2>&1
+popd > /dev/null 2>&1
+echo "[INFO] Done"
```
```suggestion
echo "[INFO] Done"

```

vsellier replied:
```suggestion
echo "[INFO] $(display_date)"
echo "[INFO] Done"
```

vsellier commented on srv/exo/bin/_dumpData.sh:
```diff
@@ -0,0 +1,27 @@
+
+echo "[INFO] ======================================="
+echo "[INFO] = Compressing ${PLF_NAME} data into ${PLF_BCK_DIR}/${PLF_NAME}-data-${SCRIPT_DATE}.tar.bz2 ..."
+echo "[INFO] ======================================="
```
```suggestion
echo "[INFO] ======================================="
echo "[INFO] $(display_date)"
```

vsellier commented on srv/exo/bin/_dumpDatabase.sh:
```diff
@@ -0,0 +1,29 @@
+#!/bin/bash -eu
```
This file should be named ``_dumpMysqlDatabase.sh``

vsellier commented on srv/exo/bin/_startDataBase.sh:
```diff
@@ -0,0 +1,24 @@
+echo "[INFO] Starting Database server on ${HOSTNAME}..."
+echo "[INFO] ======================================="
+
+if [ -e /lib/systemd/system/mysql.service ]; then
```
Is there a particular reason to test the presence of this file ?
the script should failed if the systemctl command failed

vsellier commented on srv/exo/bin/_startElasticSearch.sh:
```diff
@@ -0,0 +1,24 @@
+echo "[INFO] Starting ElasticSearch server on ${HOSTNAME}..."
+echo "[INFO] ======================================="
+
+if [ -e /usr/lib/systemd/system/elasticsearch.service ]; then
```
same as https://github.com/exoplatform/monitoring-stack/pull/3/files#diff-af3aeeda0bce4bba7807bdefeabcbb44R18

vsellier commented on srv/exo/bin/_startMongo.sh:
```diff
@@ -0,0 +1,24 @@
+echo "[INFO] Starting MongoDB server on ${HOSTNAME}..."
+echo "[INFO] ======================================="
+
+if [ -e /lib/systemd/system/mongod.service ]; then
```
same as https://github.com/exoplatform/monitoring-stack/pull/3/files#diff-af3aeeda0bce4bba7807bdefeabcbb44R18

vsellier commented on srv/exo/bin/_startPLF.sh:
```diff
@@ -0,0 +1,59 @@
+  set -e
+ 
+else
+  echo "[WARN] $(display_date) ${PLF_NAME} not deployed. Cannot be started !!!"
```
We should stop in error

vsellier commented on srv/exo/bin/_stopElasticSearch.sh:
```diff
@@ -0,0 +1,26 @@
+echo "[INFO] Stopping ElasticSearch server on ${HOSTNAME}..."
+echo "[INFO] ======================================="
+
+if [ -e /usr/lib/systemd/system/elasticsearch.service ]; then
```
same as https://github.com/exoplatform/monitoring-stack/pull/3/files#diff-af3aeeda0bce4bba7807bdefeabcbb44R18

vsellier commented on srv/exo/bin/_stopElasticSearch.sh:
```diff
@@ -0,0 +1,26 @@
+  ACTION=stop  
+  sudo systemctl ${ACTION} elasticsearch
+else
+  echo "[WARN] $(display_date) ElasticSearch not deployed. Cannot be stopped !!!"
```
We should stop here

vsellier commented on srv/exo/bin/_stopMongo.sh:
```diff
@@ -0,0 +1,25 @@
+echo "[INFO] Stoping MongoDB server on ${HOSTNAME}..."
+echo "[INFO] ======================================="
+
+if [ -e /lib/systemd/system/mongod.service ]; then
```
same as https://github.com/exoplatform/monitoring-stack/pull/3/files#diff-af3aeeda0bce4bba7807bdefeabcbb44R18

vsellier commented on srv/exo/bin/_stopPLF.sh:
```diff
@@ -0,0 +1,30 @@
+    pgrep -U $(id -u) soffice | xargs kill -9
+  fi
+else
+  echo "[WARN] ${PLF_NAME} not deployed. Cannot be stopped !!!"
```
we should stop here

vsellier commented on srv/exo/bin/backup_plateforme.sh:
```diff
@@ -0,0 +1,40 @@
+
+DOWNTIME_START_TIME=$(date +%s)
+# Stop it
+ssh -i ${SCRIPT_DIR}/id_rsa ${EXO_USER}@${EXO_PLF_SERVER} ${SCRIPT_DIR}/_stopPLF.sh
```
We should rely on the system to ensure the current user can connect on the distant server without specifying the ssh key

vsellier reviewed (changes requested):
- This should be placed on a ``backup`` directory, not directly under ``srv/bin``
- The elasticsearch directory backup is missing

vsellier committed:
Update srv/exo/bin/_dumpDatabase.sh

Co-Authored-By: bourasom <omar.bouras@exoplatform.com>

vsellier committed:
Update srv/exo/bin/_dumpDatabase.sh

Co-Authored-By: bourasom <omar.bouras@exoplatform.com>

vsellier committed:
Update srv/exo/bin/_dumpDatabase.sh

Co-Authored-By: bourasom <omar.bouras@exoplatform.com>

vsellier committed:
Update srv/exo/bin/_dumpData.sh

Co-Authored-By: bourasom <omar.bouras@exoplatform.com>

bourasom committed:
Update backup PR

bourasom committed:
Abort platform startup when errors in logs

bourasom committed:
Add rsync option and es backup

vsellier commented on srv/exo/bin/_dumpElasticSearch.sh:
```diff
@@ -0,0 +1,29 @@
+echo "[INFO] ======================================="
+echo "[INFO] $(display_date)"
+pushd ${BACKUP_WORKING_DIR}/tmp_elastic > /dev/null 2>&1
+display_time tar --directory ${ELASTIC_WORKING_DIR} --use-compress-prog=pbzip2 -cpf ${BACKUP_WORKING_DIR}/tmp_elastic/${PLF_NAME}-ES-${SCRIPT_DATE}.tar.bz2 elasticsearch  
```
```suggestion
display_time tar --directory ${ELASTIC_WORKING_DIR} --use-compress-prog=pbzip2 -cpf ${BACKUP_WORKING_DIR}/tmp_elastic/${PLF_NAME}-es-${SCRIPT_DATE}.tar.bz2 elasticsearch  
```

bourasom replied:
Done

vsellier commented on srv/exo/bin/_startDataBase.sh:
```diff
@@ -0,0 +1,20 @@
+#!/bin/bash -eu
```
Should be name _startDatabase.sh

bourasom replied:
Done

vsellier commented on srv/exo/bin/_startElasticSearch.sh:
```diff
@@ -0,0 +1,20 @@
+#!/bin/bash -eu
```
should be named _startElasticsearch.sh

bourasom replied:
Done

vsellier commented on srv/exo/bin/_dumpData.sh:
```diff
@@ -0,0 +1,29 @@
+SCRIPT_DIR="$( cd -P "$( dirname "${BASH_SOURCE[0]}" )" && pwd )" 
+
+# Load env settings
+source ${SCRIPT_DIR}/_setenv-template.sh
```
```suggestion
source ${SCRIPT_DIR}/setenv.sh
```

bourasom replied:
Done

vsellier commented on srv/exo/bin/_dumpElasticSearch.sh:
```diff
@@ -0,0 +1,29 @@
+SCRIPT_DIR="$( cd -P "$( dirname "${BASH_SOURCE[0]}" )" && pwd )" 
+
+# Load env settings
+source ${SCRIPT_DIR}/_setenv-template.sh
```
```suggestion
source ${SCRIPT_DIR}/setenv.sh
```

bourasom replied:
Done

vsellier commented on srv/exo/bin/_dumpMongoDb.sh:
```diff
@@ -0,0 +1,33 @@
+SCRIPT_DIR="$( cd -P "$( dirname "${BASH_SOURCE[0]}" )" && pwd )" 
+
+# Load env settings
+source ${SCRIPT_DIR}/_setenv-template.sh
```
```suggestion
source ${SCRIPT_DIR}/setenv.sh
```

vsellier commented on srv/exo/bin/_dumpMysqlDatabase.sh:
```diff
@@ -0,0 +1,31 @@
+SCRIPT_DIR="$( cd -P "$( dirname "${BASH_SOURCE[0]}" )" && pwd )" 
+
+# Load env settings
+source ${SCRIPT_DIR}/_setenv-template.sh
```
```suggestion
source ${SCRIPT_DIR}/setenv.sh
```

vsellier commented on srv/exo/bin/_startDataBase.sh:
```diff
@@ -0,0 +1,20 @@
+SCRIPT_DIR="$( cd -P "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+# Load env settings
+source ${SCRIPT_DIR}/_setenv-template.sh
```
```suggestion
source ${SCRIPT_DIR}/setenv.sh
```

vsellier commented on srv/exo/bin/_startElasticSearch.sh:
```diff
@@ -0,0 +1,20 @@
+SCRIPT_DIR="$( cd -P "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+# Load env settings
+source ${SCRIPT_DIR}/_setenv-template.sh
```
```suggestion
source ${SCRIPT_DIR}/setenv.sh
```

vsellier commented on srv/exo/bin/_startMongo.sh:
```diff
@@ -0,0 +1,20 @@
+SCRIPT_DIR="$( cd -P "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+# Load env settings
+source ${SCRIPT_DIR}/_setenv-template.sh
```
```suggestion
source ${SCRIPT_DIR}/setenv.sh
```

vsellier commented on srv/exo/bin/_startPLF.sh:
```diff
@@ -0,0 +1,66 @@
+SCRIPT_DIR="$( cd -P "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+# Load env settings
+source ${SCRIPT_DIR}/_setenv-template.sh
```
```suggestion
source ${SCRIPT_DIR}/setenv.sh
```

vsellier commented on srv/exo/bin/_stopDatabase.sh:
```diff
@@ -0,0 +1,25 @@
+SCRIPT_DIR="$( cd -P "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+# Load env settings
+source ${SCRIPT_DIR}/_setenv-template.sh
```
```suggestion
source ${SCRIPT_DIR}/setenv.sh
```

vsellier commented on srv/exo/bin/_stopElasticSearch.sh:
```diff
@@ -0,0 +1,26 @@
+SCRIPT_DIR="$( cd -P "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+# Load env settings
+source ${SCRIPT_DIR}/_setenv-template.sh
```
```suggestion
source ${SCRIPT_DIR}/setenv.sh
```

vsellier commented on srv/exo/bin/_stopMongo.sh:
```diff
@@ -0,0 +1,24 @@
+SCRIPT_DIR="$( cd -P "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+# Load env settings
+source ${SCRIPT_DIR}/_setenv-template.sh
```
```suggestion
source ${SCRIPT_DIR}/setenv.sh
```

vsellier commented on srv/exo/bin/_stopPLF.sh:
```diff
@@ -0,0 +1,27 @@
+SCRIPT_DIR="$( cd -P "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+# Load env settings
+source ${SCRIPT_DIR}/_setenv-template.sh
```
```suggestion
source ${SCRIPT_DIR}/setenv.sh
```

vsellier commented on srv/exo/bin/backup_plateforme.sh:
```diff
@@ -0,0 +1,46 @@
+SCRIPT_START_TIME=$(date +%s)
+
+# Load env settings
+source ${SCRIPT_DIR}/_setenv-template.sh
```
```suggestion
source ${SCRIPT_DIR}/setenv.sh
```

vsellier commented on srv/exo/bin/start_plateforme.sh:
```diff
@@ -0,0 +1,22 @@
+SCRIPT_DIR="$( cd -P "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+# Load env settings
+source ${SCRIPT_DIR}/_setenv-template.sh
```
```suggestion
source ${SCRIPT_DIR}/setenv.sh
```

vsellier commented on srv/exo/bin/stop_plateforme.sh:
```diff
@@ -0,0 +1,26 @@
+SCRIPT_DIR="$( cd -P "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+# Load env settings
+source ${SCRIPT_DIR}/_setenv-template.sh
```
```suggestion
source ${SCRIPT_DIR}/setenv.sh
```

vsellier commented on srv/exo/bin/_setenv-template.sh:
```diff
@@ -0,0 +1,42 @@
+#!/bin/sh -ue
```
We should name it setenv.sh to simplify the process for the client point of view

vsellier replied:
I mean setenv-template.sh without the _

vsellier commented on srv/exo/bin/_startDataBase.sh:
```diff
@@ -0,0 +1,20 @@
+echo "[INFO] ======================================="
+echo ""
+ACTION=start  
+sudo systemctl ${ACTION} mysql
```
We should at least checking the service status if up after calling the start.
It would be best to check that mysql is launched and available before the return.

vsellier commented on srv/exo/bin/_startMongo.sh:
```diff
@@ -0,0 +1,20 @@
+echo "[INFO] ======================================="
+echo ""
+ACTION=start  
+sudo systemctl ${ACTION} mongod
```
We should at least checking the service status if up after calling the start.
It would be best to check that mongo is launched and available before the return.

vsellier commented on srv/exo/bin/start_plateforme.sh:
```diff
@@ -0,0 +1,22 @@
+#!/bin/bash -eu
```
should be name startPlatform.sh

vsellier commented on srv/exo/bin/stop_plateforme.sh:
```diff
@@ -0,0 +1,26 @@
+
```
should be name stopPlatform.sh

vsellier replied:
```suggestion
```

vsellier commented on srv/exo/bin/backup_plateforme.sh:
```diff
@@ -0,0 +1,46 @@
+#!/bin/bash -eu
```
should be named backupPlatform.sh

vsellier commented on srv/exo/bin/_stopDatabase.sh:
```diff
@@ -0,0 +1,25 @@
+echo "[INFO] Stopping Database server on ${HOSTNAME}..."
+echo "[INFO] ======================================="
+
+if [ -e /lib/systemd/system/mysql.service ]; then
```
You should rely on a systemctl command to detect the declaration of the service
like ``systemctl list-units mysql.service``, it will avoid to depend on the way the service is installed

vsellier commented on srv/exo/bin/_setenv-template.sh:
```diff
@@ -0,0 +1,42 @@
+# 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+#
+
+# Directories
```
you should reorganise the variable declarations to match the sections

vsellier commented on srv/exo/bin/_setenv-template.sh:
```diff
@@ -0,0 +1,42 @@
+# Directories
+PLF_SRV_DIR=/srv/exo
+PLF_LOG_DIR=/srv/exo/current/logs
+PLF_NAME=exo
```
WDYT of ``EXO_SERVICE_NAME`` ?

vsellier commented on srv/exo/bin/_setenv-template.sh:
```diff
@@ -0,0 +1,42 @@
+BACKUP_DIR=/srv/backup
+BACKUP_WORKING_DIR=/var/backup
+#Database Backup
+PLF_DATABASE_NAME=exo
```
WDYT of ``EXO_DATABASE_NAME`` ?

vsellier commented on srv/exo/bin/_setenv-template.sh:
```diff
@@ -0,0 +1,42 @@
+#Data Backup
+DATA_WORKING_DIR=/srv/data/exo
+#Mongo Backup
+EXO_CHAT_MONGODB_NAME=chat
```
WDYT of ``EXO_CHAT_DATABASE_NAME`` ?

vsellier commented on srv/exo/bin/_setenv-template.sh:
```diff
@@ -0,0 +1,42 @@
+#Mongo Backup
+EXO_CHAT_MONGODB_NAME=chat
+#Elastic Backup
+ELASTIC_WORKING_DIR=/srv/data/
```
Why not using the global BACKUP_WORKING_DIR variable ?

vsellier commented on srv/exo/bin/_setenv-template.sh:
```diff
@@ -0,0 +1,42 @@
+#Elastic Backup
+ELASTIC_WORKING_DIR=/srv/data/
+#DOWNLOAD BACKUP
+DOWNLOAD_BACKUP=false
```
perhaps a little explanation of this variable purpose could help to make the right choice ;)

vsellier commented on srv/exo/bin/_dumpData.sh:
```diff
@@ -0,0 +1,29 @@
+#!/bin/bash -eu
```
WDYT to name this script ``dumpExoData.sh`` ?

vsellier commented on srv/exo/bin/_dumpData.sh:
```diff
@@ -0,0 +1,29 @@
+echo "[INFO] ======================================="
+echo "[INFO] $(display_date)"
+pushd ${BACKUP_WORKING_DIR}/tmp_data > /dev/null 2>&1
+display_time tar --directory ${DATA_WORKING_DIR} --use-compress-prog=pbzip2 -cpf ${BACKUP_WORKING_DIR}/tmp_data/${PLF_NAME}-data-${SCRIPT_DATE}.tar.bz2 data  
```
We should be sure pbzip2 is installed before using it

vsellier commented on srv/exo/bin/_dumpData.sh:
```diff
@@ -0,0 +1,29 @@
+
+# Initialize working directory
+mkdir -p ${BACKUP_WORKING_DIR}/tmp_data
+rm -rf ${BACKUP_WORKING_DIR}/tmp_data/*
```
You should add the ``v`` option to help for a potential need to debug

vsellier commented on srv/exo/bin/_dumpData.sh:
```diff
@@ -0,0 +1,29 @@
+# Load common functions
+source ${SCRIPT_DIR}/_functions.sh
+
+SCRIPT_DATE=`date "+%Y-%m-%d-%H%M%S"`
```
You should compute the script date value on the top level script to have a consistent backup date between all the archives (same observation for all the scripts)

vsellier commented on srv/exo/bin/_dumpMongoDb.sh:
```diff
@@ -0,0 +1,33 @@
+echo "[INFO] $(display_date)"
+
+display_time mongodump -o ${BACKUP_WORKING_DIR}/tmp_mongo --db=${EXO_CHAT_MONGODB_NAME}
+#display_time sudo chown -R exo:exo ${BACKUP_WORKING_DIR}/tmp_mongo
```
```suggestion
```

bourasom replied:
Done

vsellier commented on srv/exo/bin/_dumpMongoDb.sh:
```diff
@@ -0,0 +1,33 @@
+
+display_time mongodump -o ${BACKUP_WORKING_DIR}/tmp_mongo --db=${EXO_CHAT_MONGODB_NAME}
+#display_time sudo chown -R exo:exo ${BACKUP_WORKING_DIR}/tmp_mongo
+
```
You should create an tar of the backuped data and cleanup the wordking dir after

bourasom replied:
Done

vsellier commented on srv/exo/bin/_dumpMysqlDatabase.sh:
```diff
@@ -0,0 +1,31 @@
+echo "[INFO] ======================================="
+echo "[INFO] $(display_date)"
+
+display_time sudo mysqldump ${DUMP_OPTIONS} ${PLF_DATABASE_NAME} > ${BACKUP_WORKING_DIR}/tmp_db/${PLF_DATABASE_NAME}-dumpDataBase.sql
```
You should compress the sql dump in the fly

bourasom replied:
Done

vsellier commented on srv/exo/bin/backup_plateforme.sh:
```diff
@@ -0,0 +1,46 @@
+ssh ${EXO_USER}@${EXO_PLF_SERVER} ${SCRIPT_DIR}/_startPLF.sh
+DOWNTIME_END_TIME=$(date +%s)  
+if [ ${DOWNLOAD_BACKUP} ]; then
+    rsync -avz ${EXO_USER}@${EXO_PLF_SERVER}:${BACKUP_WORKING_DIR}/tmp_data/* ${BACKUP_DIR} 
```
the -z option on all the rsync command is not useful as bz2 file should be transferred

bourasom replied:
Done

vsellier commented on srv/exo/bin/backup_plateforme.sh:
```diff
@@ -0,0 +1,46 @@
+# Start it
+ssh ${EXO_USER}@${EXO_PLF_SERVER} ${SCRIPT_DIR}/_startPLF.sh
+DOWNTIME_END_TIME=$(date +%s)  
+if [ ${DOWNLOAD_BACKUP} ]; then
```
what if the DOWNLOAD_BACKUP is ``false`` ?

If 'm not wrong this option is here to retrieve the backup files from several servers when there is several.
WDYT to rsync the remote dirs ``${EXO_USER}@${EXO_DB_SERVER}:${BACKUP_WORKING_DIR}/...`` when ``DOWNLOAD_BACKUP=true`` and to only rsync the local ``${BACKUP_WORKING_DIR}/...`` when ``DOWNLOAD_BACKUP=false``  ?

bourasom replied:
Done

vsellier reviewed (changes requested):
* You should centralize the systemd services start and stop in a generic management script to have the same controls (service declaration, command and service status) for all of them

bourasom committed:
update backup scripts

bourasom committed:
generic systemd cmd function

bourasom commented:
>     * You should centralize the systemd services start and stop in a generic management script to have the same controls (service declaration, command and service status) for all of them

Done

bourasom committed:
local rsync in case of no backup download

vsellier commented on srv/exo/bin/_startPLF.sh:
```diff
@@ -0,0 +1,66 @@
+  set +e
+  while [ true ];
+  do
+    if grep -q "ERROR" ${PLF_LOG_DIR}/platform.log; then
```
I just realised this is really risky to stop the server if an error occurs on the logs during the startup.
There's a good chance the service never succeed to restart ;)

Commit:
code format

Commit:
Improve start stop commamds

- centralize the service detection on _functions.sh
- Remove an unnecessary variable containing the action
- Don't stop anymore tomcat if an error is detected during the startup
- Rename _*PLF.sh _*eXo.sh

Commit:
Don't stop eXo if an ERROR log appears on the startup logs

Commit:
Improve exo backup

- ensure we go back to the initial directory at the end
- ensure the backup is made from the n-1 level and don't assume the data dir is name data

Commit:
Improve elasticsearch backup

- use elasticsearch instead of elastic
- respect the full path for the backup

Commit:
Rename chat database parameter

Commit:
Mysql backup changes

- Rename database name paramter
- Rename finale dump to respect usual template

Commit:
translate start/stop script names

Commit:
Use the same date for all the archives

bourasom committed:
Add drop database instruction in the dump

bourasom committed:
Add restore scripts

bourasom committed:
update retore_plateform script

bourasom committed:
Some cleanup in restore

Commit:
Fix typo on script names

bourasom committed:
change from drop-database to drop-table in db dump option

Commit:
one more typo

Commit:
Set executable permission

Commit:
Fix eXo startup detection

Commit:
Fix one mor typo

Commit:
utility script to install/synchonize all the servers

Commit:
Elasticsearch restore

Commit:
Fix remote and local backup sync

Commit:
rename _setenv-template.sh

Commit:
improve mongo restoration

Commit:
mysql restore

Commit:
exo data restore

Commit:
allow is-active action on sudoers

Commit:
Add a backup retention delay

Commit:
Automatically retrieve ssh server fingerprint

Commit:
Fix eXo startup when it's the first start of a day

Commit:
fix frenglish typo

Commit:
fix archive transfer during the restore

Commit:
use more convenient directory

bourasom committed:
ITOP-4405: change cmnd_alias for sudoers in case of same server

Commit:
don't expect the exo data to be in a directory named data

Commit:
allow to backup locally without ssh usage

Commit:
allow to restore locally without using ssh

Commit:
TASK-23980 #6 Warmup the server after a restart (#7)
